protocol/src/types/cstring.rs

```diff
@@ -0,0 +1,61 @@
+use std::ffi::CString;
+use std::io::prelude::{Read, Write};
+use {hint, Error, Parcel, Settings};
+
+impl Parcel for CString {
+    const TYPE_NAME: &'static str = "CString";
+
+    fn read_field(
+        read: &mut dyn Read,
+        settings: &Settings,
+        _hints: &mut hint::Hints,
+    ) -> Result<Self, Error> {
+        let mut result = Vec::new();
+        loop {
```

Comment:
This loop is a possible denial of service vector due to the fact that the peer could simply never send a null byte, which would block the program.

I was going to suggest fixing it, but I spotted #14 and noted that most of the other builtin Parcel implementations have the same DoS vector and a proper API needs to be designed to handle the validations in general and relaxation of them when required.

**recommendation**: add a link to #14 to a comment before the `loop`, say something like `this logic is susceptible to DoS attacks and will be fixed by https://github.com/dylanmckay/protocol/issues/14`